packages/rainbowkit/src/wallets/walletConnectors/coreWallet/coreWallet.ts

```diff
@@ -0,0 +1,148 @@
+/* eslint-disable sort-keys-fix/sort-keys-fix */
+import type { InjectedConnectorOptions } from '@wagmi/core/connectors/injected';
+import { InjectedConnector } from 'wagmi/connectors/injected';
+import { WindowProvider } from 'wagmi/dist/window';
+import { Chain } from '../../../components/RainbowKitProvider/RainbowKitChainContext';
+import { Wallet } from '../../Wallet';
+import { getWalletConnectConnector } from '../../getWalletConnectConnector';
+
+declare global {
+  interface Window {
+    evmproviders: Record<string, WindowProvider>;
+    avalanche: WindowProvider;
+  }
+}
+
+export interface CoreWalletOptions {
+  projectId?: string;
+  chains: Chain[];
+}
+
+function getCoreWalletInjectedProvider(): WindowProvider | undefined {
+  const injectedProviderExist =
+    typeof window !== 'undefined' && typeof window.ethereum !== 'undefined';
+
+  // No injected providers exist.
+  if (!injectedProviderExist) {
+    return;
+  }
+
+  // Core implements EIP-5749 and creates the window.evmproviders
+  if (window['evmproviders']?.['core']) {
+    return window['evmproviders']?.['core'];
+  }
+
+  // Core was injected into window.avalanche.
+  if (window.avalanche) {
+    return window.avalanche;
+  }
+
+  // Core was injected into window.ethereum.
+  if (
+    typeof window !== 'undefined' &&
+    typeof window.ethereum !== 'undefined' &&
+    window.ethereum.isAvalanche === true
+  ) {
+    return window.ethereum;
+  }
+}
+
+export const coreWallet = ({
+  chains,
+  projectId,
+  ...options
+}: CoreWalletOptions & InjectedConnectorOptions): Wallet => {
+  const isCoreInjected = Boolean(getCoreWalletInjectedProvider());
+
+  const shouldUseWalletConnect = !isCoreInjected;
+  return {
+    id: 'core',
+    name: 'Core',
+    iconUrl: async () => (await import('./coreWallet.svg')).default,
+    iconBackground: '#1A1A1C',
+    installed: !shouldUseWalletConnect ? isCoreInjected : undefined,
+    downloadUrls: {
+      android: 'https://play.google.com/store/apps/details?id=com.avaxwallet',
+      ios: 'https://apps.apple.com/us/app/core-wallet/id6443685999',
+      mobile: 'https://core.app/?downloadCoreMobile=1',
+      qrCode: 'https://core.app/?downloadCoreMobile=1',
+      chrome:
+        'https://chrome.google.com/webstore/detail/core-crypto-wallet-nft-ex/agoakfejjabomempkjlepdflaleeobhb',
+      browserExtension: 'https://extension.core.app/',
+    },
+    createConnector: () => {
+      const connector = shouldUseWalletConnect
+        ? getWalletConnectConnector({ projectId, chains })
+        : new InjectedConnector({
+            chains,
+            options: {
+              name: 'Core',
```

Comment:
`name` should be inferred from Wagmi. The mismatch should be addressed in Wagmi first. https://github.com/wagmi-dev/references/blob/746df4f7f0b19dd8ca59e0cc7b6e53aa296bc767/packages/connectors/src/utils/getInjectedName.ts#LL8C6-L8C6